### DISCUSSION TITLE
MoviePy TextClip Custom Font Not Recognized (Python) - Windows 10 - ImageMagick-7.1.1-Q16-HDRI

### DISCUSSION THREAD

**Discussion by makseem77** [ImageMagick/ImageMagick#6468] [Help]
### ImageMagick version

ImageMagick-7.1.1-Q16-HDRI

### Operating system, version and so on

Windows 10

### Description

for some reason, I can't use custom fonts when creating a MoviePy TextClip. It says in their documentation that it should be to use all the fonts installed on my system, but when I run this little code snippet in Python (to get all the possible TextClip fonts):

```
from moviepy.editor import TextClip

print(TextClip.list("font"))
```

It prints a list of all the basic fonts that you get when you install Windows, but none of the custom fonts that I installed myself. (And that I can see in the Windows/Fonts directory)

I did some research and it seems that it might be related with ImageMagick, but apparently, on Windows, the fonts should be handled without the user having to do anything. I tried reinstalling ImageMagick, but it didn't solve the issue. I also found a post talking about editing the type-windows.xml file in the ImageMagick config, but when I open it, I see a list of fonts looking like this:

```
...

 <type name="Arial-Special-G2-Italic" fullname="Arial Special G2 Italic" family="Arial Special G2" weight="400" style="italic" stretch="normal" glyphs="/usr/share/fonts/msttcore/msgeoai2.ttf"/>
  <type name="Bookman-Old-Style" fullname="Bookman Old Style" family="Bookman Old Style" weight="400" style="normal" stretch="normal" glyphs="/usr/share/fonts/msttcore/bkmnos.ttf"/>
  <type name="Bookman-Old-Style-Bold" fullname="Bookman Old Style Bold" family="Bookman Old Style" weight="700" style="normal" stretch="normal" glyphs="/usr/share/fonts/msttcore/bookosb.ttf"/> 

...
```

But the fonts do not correspond with the one I see when I print the list from TextClip.list + I didn't find anything on how to manually add a font there so I gave up on this lead.

If anyone has some ideas on what I'm doing wrong, it would be greatly appreciated !

**Comment by Lcorinst** [comment]
Right click on your ttf font file, select install for all users, then TextClip.list("font") will see the name of the newly installed font in the list, use it in the front parameter of TextClip

**Reply by lilaboc** [reply]
this worked for me, thanks!

**Reply by moha6767** [reply]
I dont know but it doesnt work.

**Reply by csongorszasz** [reply]
It worked for me too. 
Make sure to install "FOR ALL USERS".

**Comment by KrychaTech** [comment]
@Lcorinst your solution does not work.

**Comment by fmw42** [comment]
Perhaps post to MoviePy, not Imagemagick.
